src/mesh/index_derivs.cxx

```diff
@@ -2837,7 +2837,7 @@ const Field3D Mesh::indexFDDX(const Field3D &v, const Field3D &f, CELL_LOC outlo
     }
   } else if (this->xstart == 1) {
```

Comment:
Again not something you've changed, just something I've noticed -- I think this only needs to be else if we assume `xstart>0` is asserted somewhere.

Reply:
I don't think xstart>0 is actually required. It should be possible to run 2d simulations with no guard cells in the 3rd direction. Then this test is needed so that an exception is thrown if the user tries to take a derivative in the 3rd direction.

Reply:
Currently this does throw if `xstart == 0` (the final `else` clause), so I agree we should move that to an `ASSERT` near the top.

Reply:
I know in some routines there's an "early exit" check that says, e.g. `if nz == 1 {return 0;}`, maybe we that here.